zulipterminal/ui_tools/boxes.py

```diff
@@ -96,18 +96,23 @@ def stream_box_view(self, caption: str='', title: str='') -> None:
 
     def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
         if text.startswith('@_'):
-            return self.autocomplete_mentions(text, state, '@_')
+            typeahead = self.autocomplete_mentions(text, '@_')
         elif text.startswith('@'):
-            return self.autocomplete_mentions(text, state, '@')
+            typeahead = self.autocomplete_mentions(text, '@')
         elif text.startswith('#'):
-            return self.autocomplete_streams(text, state)
+            typeahead = self.autocomplete_streams(text)
         elif text.startswith(':'):
-            return self.autocomplete_emojis(text, state)
+            typeahead = self.autocomplete_emojis(text)
```

Comment:
We may be able to simplify this block further - it seems rather repetitive - though we don't have to do it in this PR necessarily.

Reply:
I'll probably look at this in the footer PR, some of the changes made there could affect the refactor here.